utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -0,0 +1,68 @@
+package org.oppia.android.util.logging.performancemetrics
+
+import android.app.ActivityManager
+import android.content.Context
+import android.net.TrafficStats
+import org.oppia.android.app.model.OppiaMetricLog
+import java.io.File
+import javax.inject.Inject
+
+/** Utility to extract performance metrics from the underlying Android system. */
+class PerformanceMetricsAssessorImpl @Inject constructor(
+  private val context: Context
+) : PerformanceMetricsAssessor {
+
+  private val activityManager: ActivityManager by lazy {
+    context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+  }
+
+  override fun getApkSize(): Long {
+    val apkPath =
+      context.packageManager.getPackageInfo(context.packageName, 0).applicationInfo.sourceDir
+    return File(apkPath).length() / 1024
+  }
+
+  override fun getUsedStorage(): Long {
+    val permanentStorageUsage = context.filesDir?.listFiles()?.map { it.length() }?.sum()
+    val cacheStorageUsage = context.cacheDir?.listFiles()?.map { it.length() }?.sum()
+    return (cacheStorageUsage?.let { permanentStorageUsage?.plus(it) }) ?: 0L
+  }
+
+  override fun getTotalSentBytes(): Long = TrafficStats.getUidTxBytes(context.applicationInfo.uid)
+
+  override fun getTotalReceivedBytes(): Long =
+    TrafficStats.getUidRxBytes(context.applicationInfo.uid)
+
+  override fun getTotalPssUsed(): Long {
+    val pid = ActivityManager.RunningAppProcessInfo().pid
+    activityManager.runningAppProcesses
```

Comment:
Is this being used anymore? It seems to be a hanging expression.

Reply:
removed.